### PR TITLE
feat(atomic): allow to customise suggestions icons

### PR DESCRIPTION
**Pull request by liviarett**
https://coveord.atlassian.net/browse/KIT-1626

### CODE REVIEW

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx [hunk 1]
```diff
@@ -29,6 +29,12 @@ export class AtomicSearchBoxRecentQueries {
 
   @State() public error!: Error;
 
+  /**
+   * The icon that will be passed down to `atomic-icon` to be shown next to each suggestion on the list.
+   * Defaults to atomic icon. Setting it to an empty string hides it.
+   */
+  @Prop() public icon?: string;
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx:36]: It shows/hides itself when there other types of suggestions along with it. It's what UX deemed best, there should always be at least a default icon

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx [hunk 2]
```diff
@@ -97,8 +108,13 @@ export class AtomicSearchBoxRecentQueries {
     }
 
     const query = this.bindings.searchBoxController.state.value;
+    console.log(
+      'this.bindings.searchBoxController.state:',
+      this.bindings.searchBoxController.state
+    );
     const hasQuery = query !== '';
     const max = hasQuery ? this.maxWithQuery : this.maxWithoutQuery;
+    console.log('max:', max);
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx:117]:
```suggestion
```

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx [hunk 3]
```diff
@@ -97,8 +108,13 @@ export class AtomicSearchBoxRecentQueries {
     }
 
     const query = this.bindings.searchBoxController.state.value;
+    console.log(
+      'this.bindings.searchBoxController.state:',
+      this.bindings.searchBoxController.state
+    );
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx:114]:
```suggestion
```

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx [hunk 4]
```diff
@@ -45,6 +51,11 @@ export class AtomicSearchBoxQuerySuggestions {
     }
   }
 
+  get Icon() {
+    if (this.icon === '') return null;
+    return this.icon || SearchIcon;
```
Comment (btaillon) [inline packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx:56]: I prefer `??` when we want to fallback from `null` or` undefined`.

```suggestion
    return this.icon ?? SearchIcon;
```

**Comment by ThibodeauJF** [comment]
An alternative solution would be to use a [part](https://stenciljs.com/docs/styling#css-parts)
We already have the `atomic-icon` component that our users can use for icons, they could also put their own images or svgs directly

```
<atomic-search-box-recent-queries>
 <svg part="icon" ... />
</atomic-search-box-recent-queries>
```

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx (hunk 1 above)
Comment (liviarett) [reply packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx:36]: alright

**Comment by liviarett** [comment]
I personally prefer it as a prop, I find it clearer to understand, but if you guys prefer a part, I'll change.

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx [hunk 5]
```diff
@@ -25,6 +25,12 @@ export class AtomicSearchBoxQuerySuggestions {
 
   @State() public error!: Error;
 
+  /**
+   * The icon that will be passed down to `atomic-icon` to be shown next to each suggestion on the list.
+   * Defaults to the atomic icon.
```
Comment (olamothe) [inline packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx:30]: What is the `atomic icon` ? 🤔 It should be a bit more precise.

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx [hunk 6]
```diff
@@ -29,6 +29,12 @@ export class AtomicSearchBoxRecentQueries {
 
   @State() public error!: Error;
 
+  /**
+   * The icon that will be passed down to `atomic-icon` to be shown next to each suggestion on the list.
+   * Defaults to the atomic icon.
```
Comment (olamothe) [inline packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx:34]: Same as above.

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx (hunk 5 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx:30]: If we keep the prop, we can copy the doc from atomic-icon

```
/**
 * The SVG icon to display.
 *
 * - Use a value that starts with `http://`, `https://`, `./`, or `../`, to fetch and display an icon from a given location.
 * - Use a value that starts with `assets://`, to display an icon from the Atomic package.
 * - Use a stringified SVG to display it directly.
 */
```

Comment (liviarett) [reply packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx:30]: ah ok! I thought we wouldnt want to duplicate, and that we could link the docs! but I'll update!!

**Comment by liviarett** [comment]
Let's have a look at what it would look like to have it as a slot

File: packages/atomic/src/pages/examples/suggestions.html [hunk 7]
```diff
@@ -82,7 +82,9 @@
         <atomic-layout-section section="search">
           <atomic-search-box suggestion-timeout="1000">
             <custom-suggestions></custom-suggestions>
-            <atomic-search-box-query-suggestions></atomic-search-box-query-suggestions>
+            <atomic-search-box-query-suggestions
+              icon="../../assets/magnifier.svg"
```
Comment (ThibodeauJF) [inline packages/atomic/src/pages/examples/suggestions.html:86]: using another icon from the assets we already have could be better than adding new ones in pages/magnifier

> Use a value that starts with `assets://`, to display an icon from the Atomic package.

**Review by ThibodeauJF — APPROVED** [review]

File: packages/atomic/src/pages/examples/suggestions.html (hunk 7 above)
Comment (liviarett) [reply packages/atomic/src/pages/examples/suggestions.html:86]: I can't seem to make it work - setting it to `assets://search.svg` for example, seems to do something
Screenshot 2022-05-13 at 10 03 51
which is different from any other string:
Screenshot 2022-05-13 at 10 03 11
but still doesn't load the icon, as you can see! do you know why?

Comment (ThibodeauJF) [reply packages/atomic/src/pages/examples/suggestions.html:86]: Oh let me look...

Comment (liviarett) [reply packages/atomic/src/pages/examples/suggestions.html:86]: ermmm that's embarrassing! sorry 😬  I tried a couple and must have been unlucky to choose only white ones!!

**Pull request merged**